lib/ElementHandle.js

```diff
@@ -103,6 +103,22 @@ class ElementHandle {
     const objectId = this._remoteObject.objectId;
     return this._client.send('DOM.setFileInputFiles', { objectId, files });
   }
+
+  /**
+   * @return {!Promise<Object>}
+   */
+  async boundingBox() {
+    const layoutMetrics = await this._client.send('Page.getLayoutMetrics');
+    const boxModel = await this._client.send('DOM.getBoxModel', { objectId: this._remoteObject.objectId });
```

Comment:
Yes, as you pointed out, we need to consider elements inside iframes and this logic you suggest applies only to the frame containing the element. Using a while loop to get the parent frame for offsets might not work because of CORS policy which makes the method inconsistant.
On the other hand using the DevTools API gives us the ability to inspect the window without browser JavaScript constraints.
So it's a matter of deciding if we want to support nested element bounding boxes or not (or we do but only for same domain frames).